lib/spack/spack/filesystem_view.py

```diff
@@ -712,7 +712,13 @@ def add_specs(self, *specs: spack.spec.Spec) -> None:
         def skip_list(file):
             return os.path.basename(file) == spack.store.STORE.layout.metadata_dir
 
-        visitor = SourceMergeVisitor(ignore=skip_list)
+        # Determine if the root is on a case-insensitive filesystem
+        sentinel_name = ".sentinel"
+        with open(os.path.join(self._root, sentinel_name), "w") as f:
+            f.write("sentinel")
```

Comment:
The current implementation of views is to always create a fresh root, but people have requested to mutate an existing path. So, better to not assume `.sentinel` is a file you can write to. Instead use `tempfile.mkstemp`.

Also no need to `write`, you can do `pass`.